libtransact/src/context/mod.rs

```diff
@@ -20,25 +20,74 @@
 pub type ContextId = [u8; 16];
 
 mod error;
+mod executer_agent;
 pub mod manager;
+mod scheduler_agent;
```

Comment:
I'm not a fan of the scheduler_agent/executer_agent naming as it names these modules in terms of the external modules they may interact with -- they could be used by other things in the future, so it would be better to use context-centric terms.

We could rename these later in a different PR if we think of better names.